proc/variables.go

```diff
@@ -28,20 +28,47 @@ const (
 
 // Represents a variable.
 type Variable struct {
-	Addr      uintptr
-	Name      string
-	Value     string
-	Type      string
-	dwarfType dwarf.Type
-	thread    *Thread
-
-	Len       int64
-	Cap       int64
+	Addr       uintptr
+	Name       string
+	DwarfType  dwarf.Type
+	RealType   dwarf.Type
+	SimpleType SimpleType
+	thread     *Thread
+
+	Value interface{}
+
+	Len int64
+	Cap int64
+
 	base      uintptr
 	stride    int64
 	fieldType dwarf.Type
+
+	Children []Variable
+
+	loaded     bool
+	Unreadable error
 }
 
+type SimpleType string
+
+const (
```

Comment:
I need a dwarf.Entry to do that and when I dereference pointers, access struct members and index arrays I don't have it; I only have a dwarf.Type and there is no easy way to go from that to a type entry. 

To do what you suggest I would have to convert the dwarf.Type into a String and search for its entry in newVariable. Should I do that?